src/components/PeopleFilters/PeopleFilters.tsx

```diff
@@ -0,0 +1,123 @@
+import cn from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+import { SexFilter } from '../../types/SexFilter';
+import { SexFilterItem } from '../SexFilterItem/SexFilterItem';
+import { SearchLink } from '../SearchLink';
+
+const centuryFilters = ['16', '17', '18', '19', '20'];
+const sexFilters = [
+  { title: 'All', value: SexFilter.ALL },
+  { title: 'Male', value: SexFilter.MALE },
+  { title: 'Female', value: SexFilter.FEMALE },
+];
+
+export const PeopleFilters: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  // const { personSlug } = useParams();
+
+  const centuries = searchParams.getAll('centuries') || [];
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || SexFilter.ALL;
+
+  const handleCenturyToggle = (century: string) => {
+    const newCenturies = centuries.includes(century)
+      ? centuries.filter((centuryElem) => centuryElem !== century)
+      : [...centuries, century];
+
+    return { centuries: newCenturies };
+  };
+
+  const handleOnQueryChange = (newQuery: string) => {
+    const params = new URLSearchParams(searchParams);
+
+    if (!newQuery) {
+      params.delete('query');
+      setSearchParams(params);
+
+      return;
+    }
+
+    params.set('query', newQuery);
+
+    setSearchParams(params);
+  };
+
+  const handleOnSexChange = (newSex: SexFilter) => {
+    return { sex: newSex };
+  };
+
+  return (
+    <nav className="panel">
+      <p className="panel-heading">Filters</p>
+
+      <p className="panel-tabs" data-cy="SexFilter">
+        {sexFilters.map(({ title, value }) => (
+          <SexFilterItem
+            title={title}
+            value={value}
+            activeSexFilter={sex as SexFilter}
+            onSexChange={handleOnSexChange}
+            key={value}
+          />
+        ))}
+      </p>
+
+      <div className="panel-block">
+        <p className="control has-icons-left">
+          <input
+            data-cy="NameFilter"
+            type="search"
+            className="input"
+            placeholder="Search"
+            value={query}
+            onChange={(event) => handleOnQueryChange(event.currentTarget.value)}
+          />
```

Comment:
You can just write `{handleOnQueryChange}` and in the function declare newQuery variable with `event.target.value`

Reply:
I understand but do you think it is really necessary?